### PR TITLE
SpriteFramesEditor Add animation searchbox

### LINKED ISSUE #2681: Search box in SpriteFrames editor interface
### Describe the project you are working on
A 2D platformer in which the player character has many different AnimatedSprite animations

### Describe the problem or limitation you are having in your project
Due to the large amount of animations for the character, finding a specific animation within the SpriteFrames resource in the editor is frustrating because there is no search functionality unlike other parts of the editor.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation
A simple search box above the scrolling list of animations that filters the animations by name, similarly to the FileSystem interface. Would allow users to find animations more quickly when there are many to filter through in the same resource.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams
<a href="https://ibb.co/fNZNTW5">mockup</a>

### If this enhancement will not be used often, can it be worked around with a few lines of script?
No, it fundamentally changes the editor layout.

### Is there a reason why this should be core and not an add-on in the asset library?
Similar search boxes are already found in the base editor offering the same functionality in different interfaces, such as the FileSystem, Scene view, and Inspector view. This would work in the exact same way, just in the SpriteFrames viewer.

### PR DESCRIPTION
**Pull request by kleonc**
`master` version of #49255. Didn't test this port.
Closes godotengine/godot-proposals#2681.

### CODE REVIEW

File: editor/plugins/sprite_frames_editor_plugin.cpp [hunk 1]
```diff
@@ -993,6 +1015,21 @@ SpriteFramesEditor::SpriteFramesEditor() {
 	hbc_animlist->add_child(remove_anim);
 	remove_anim->connect("pressed", callable_mp(this, &SpriteFramesEditor::_animation_remove));
 
+	toggle_anim_search_box = memnew(Button);
+	toggle_anim_search_box->set_flat(true);
+	toggle_anim_search_box->set_toggle_mode(true);
+	toggle_anim_search_box->set_tooltip(TTR("Show animations filter box"));
```
Comment (akien-mga) [inline editor/plugins/sprite_frames_editor_plugin.cpp:1021]:
```suggestion
	toggle_anim_search_box->set_tooltip(TTR("Show animations filter box."));
```

File: editor/plugins/sprite_frames_editor_plugin.cpp [hunk 2]
```diff
@@ -663,6 +665,22 @@ void SpriteFramesEditor::_animation_remove_confirmed() {
 	undo_redo->commit_action();
 }
 
+void SpriteFramesEditor::_animation_toggle_search_box_visibility() {
+	if (toggle_anim_search_box->is_pressed()) {
+		toggle_anim_search_box->set_tooltip(TTR("Hide animations filter box"));
+		anim_search_box->show();
+	} else {
+		toggle_anim_search_box->set_tooltip(TTR("Show animations filter box"));
```
Comment (akien-mga) [inline editor/plugins/sprite_frames_editor_plugin.cpp:673]:
```suggestion
		toggle_anim_search_box->set_tooltip(TTR("Hide animations filter box."));
		anim_search_box->show();
	} else {
		toggle_anim_search_box->set_tooltip(TTR("Show animations filter box."));
```

**Comment by akien-mga** [comment]
Would need a rebase and would be worth (re)testing to see if it still works as intended.

Otherwise the concept makes sense to me, I added the proposal to the dashboard for the next review meeting.

**Comment by kleonc** [comment]
> Would need a rebase and would be worth (re)testing to see if it still works as intended.

Rebased, works as intended (a simple search for a substring using `to_lower()`). I've made a change so now the search box is being auto-focused after it is shown (previously you'd need to click the search box after the toggle button).

**Comment by groud** [comment]
So we discussed this PR a bit during the PR meeting. Could you amend your PR with the following changes? :
- Remove the button to toggle the search LineEdit,
- Make the search LineEdit always visible, on the same line as the create and delete buttons, to the left,
- Add the "search" icon inside the LineEdit, as is it done as in most places of the editor (like in the node search bar or the property search bar).

**Comment by kleonc** [comment]
Will do.

> * Make the search LineEdit always visible, on the same line as the create and delete buttons, to the left,

To make it clear: create/delete/search (this one I'd somehow expect) or search/create/delete (this one I think you're suggesting)?

**Comment by akien-mga** [comment]
Create/delete/search IMO.

**Comment by kleonc** [comment]
Changes done, now it by default looks like this:
godot windows tools 64_W0jSqtEHFr

but it still can be shrinked to this (I intentionally didn't set a minimum size of a search box, instead the initial offset of a split container is set):
AglpXzRQH0

Search is still just a simple search for a substring using `to_lower()`.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
This will likely still be cropped for many languages which are more verbose than English (e.g. French or German).
How about just "Filter"?

**Comment by akien-mga** [comment]
Well let's keep it consistent with the rest, users of localized editors can extend the container if needed.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**